### DISCUSSION TITLE
DuckDB connect on query execute; auto disconnect

### DISCUSSION THREAD

**Discussion by ghost** [tconbeer/harlequin#556] [Ideas (Feature Requests)]
Would be nice to leave Harlequin open while running DBT against a DuckDB and immediately run a query.

DBeaver had an option for this, where you specified a number of idle seconds for auto disconnect.

**Comment by tconbeer** [comment]
I like this idea. Other than the minimal perf hit of re-establishing the connection, are there other downsides/side-effects to disconnecting?

Also, you may be interested in another solution to this problem: https://github.com/jwills/buenavista . Using a Buena Vista proxy, you can connect to a duckdb database using the `harlequin-postgres` adapter.

**Reply by ghost** [reply]
Not sure;) I don't think so. I'm only a few months into my DuckDB love affair. I'm building a replacement data warehouse stack with DBT-DuckDB as it's ETL engine, and being able to leave harlequin open in between DBT runs would save me fractional time that adds up and a better flow-state dev process. I'll give that Postgres server suggestion a spin.

I'll also crack open your code and see if I can't find my bearings on how this might be implemented. Thanks for such a great product. To be continued...
